### PR TITLE
TMA-1005: Automate rotating credentials

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: spec/integration/models/project_spec.rb [hunk 1]
```diff
@@ -163,7 +163,7 @@
   describe '#export_clone' do
     context 'when exclude_schedule is true' do
       let(:options) { { exclude_schedules: true } }
-      let(:clone) { GoodData::Project.create(title: 'project clone', client: @client, auth_token: ConnectionHelper::GD_PROJECT_TOKEN) }
+      let(:clone) { GoodData::Project.create(title: 'project clone', client: @client, auth_token: ConnectionHelper::SECRETS[:gd_project_token]) }
```
Comment (yenkins) [inline spec/integration/models/project_spec.rb]: warning: Similar code found in project_spec.rb:166, project_spec.rb:207 (mass = 32) (flay)

File: spec/integration/models/project_spec.rb [hunk 2]
```diff
@@ -204,7 +204,7 @@
   end
 
   describe '#import_clone' do
-    let(:clone) { GoodData::Project.create(title: 'import clone test', client: @client, auth_token: ConnectionHelper::GD_PROJECT_TOKEN) }
+    let(:clone) { GoodData::Project.create(title: 'import clone test', client: @client, auth_token: ConnectionHelper::SECRETS[:gd_project_token]) }
```
Comment (yenkins) [inline spec/integration/models/project_spec.rb]: warning: Similar code found in project_spec.rb:166, project_spec.rb:207 (mass = 32) (flay)

File: spec/environment/environment.rb [hunk 3]
```diff
@@ -19,7 +21,17 @@ def load(env = ENV['GD_ENV'] || 'testing')
...
+          secrets = YAML.load_file(secrets_path)
+          env_secrets = secrets[env.downcase].symbolize_keys
+          encryption_key = ENV['GD_SPEC_PASSWORD'] || ENV['BIA_ENCRYPTION_KEY']
+          env_secrets.each do |_, value|
+            decrypted = GoodData::Helpers.decrypt(value, encryption_key)
+            value.replace(decrypted)
+          end
+          GoodData::Environment::ConnectionHelper.set_const('SECRETS', env_secrets)
         rescue
+          require 'pry';binding.pry
```
Comment (yenkins) [inline spec/environment/environment.rb]: warning: Style/Semicolon: Do not use semicolons to terminate expressions. (rubocop)

Comment (yenkins) [inline spec/environment/environment.rb]: warning: Style/SpaceAfterSemicolon: Space missing after semicolon. (rubocop)

Comment (yenkins) [inline spec/environment/environment.rb]: warning: Lint/Debugger: Remove debugger entry point `binding.pry`. (rubocop)

File: spec/helpers/connection_helper.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@ class << self
         #
         # @param [String] username Optional username
         # @param [String] password Optional password
-        def create_default_connection(username = GoodData::Environment::ConnectionHelper::DEFAULT_USERNAME, password = GoodData::Environment::ConnectionHelper::DEFAULT_PASSWORD)
+        def create_default_connection(username = GoodData::Environment::ConnectionHelper::DEFAULT_USERNAME, password = GoodData::Environment::ConnectionHelper::SECRETS[:default_password])
```
Comment (yenkins) [inline spec/helpers/connection_helper.rb]: warning: Metrics/LineLength: Line is too long. [187/180] (rubocop)

File: spec/integration/command_projects_spec.rb [hunk 5]
```diff
@@ -13,7 +13,7 @@
 
     @blueprint = GoodData::Model::ProjectBlueprint.from_json("./spec/data/blueprints/test_project_model_spec.json")
     @module_blueprint = GoodData::Model::ProjectBlueprint.from_json("./spec/data/blueprints/additional_dataset_module.json")
-    @project = GoodData::Command::Project.build(:spec => @blueprint, :token => ConnectionHelper::GD_PROJECT_TOKEN, environment: ProjectHelper::ENVIRONMENT, :client => @client)
+    @project = GoodData::Command::Project.build(:spec => @blueprint, :token => ConnectionHelper::SECRETS[:gd_project_token], environment: ProjectHelper::ENVIRONMENT, :client => @client)
```
Comment (yenkins) [inline spec/integration/command_projects_spec.rb]: warning: Metrics/LineLength: Line is too long. [185/180] (rubocop)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/534/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1370/ : FAILURE in 1m 15s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1640/ : SUCCESS in 2m 25s

File: spec/environment/environment.rb [hunk 6]
```diff
@@ -15,24 +17,32 @@ class << self
...
 
+      private
+
       # Translates ci-infra branch to environment config name
       # @param branch Name of the branch we are testing against
       def branch_to_environment(branch)
         BRANCH_TO_ENVIRONMENT[branch.to_sym] || branch
       end
+
+      def decrypt_secrets(env)
```
Comment (yenkins) [inline spec/environment/environment.rb]: info: Has approx 7 statements (TooManyStatements) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/537/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1374/ : FAILURE in 1m 08s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1644/ : SUCCESS in 2m 14s

File: spec/integration/models/project_spec.rb [hunk 7]
```diff
@@ -160,65 +160,67 @@
...
 
-  describe '#export_clone' do
-    context 'when exclude_schedule is true' do
-      let(:options) { { exclude_schedules: true } }
-      let(:clone) { GoodData::Project.create(title: 'project clone', client: @client, auth_token: ConnectionHelper::GD_PROJECT_TOKEN) }
-
-      after do
-        clone.delete if clone
+  describe 'cloning' do
+    let(:clone) { GoodData::Project.create(title: 'project clone', client: @client, auth_token: ConnectionHelper::SECRETS[:gd_project_token]) }
```
Comment (yenkins) [inline spec/integration/models/project_spec.rb]: warning: Similar code found in project_spec.rb:164, project_spec.rb:208 (mass = 32) (flay)

File: spec/integration/models/project_spec.rb [hunk 8]
```diff
@@ -160,65 +160,67 @@
...
+
+        it 'raises ExportCloneError' do
+          expect { @project.export_clone }.to raise_error(GoodData::ExportCloneError)
+        end
       end
     end
 
-    context 'when export task fails' do
+    describe '#import_clone' do
+      let(:clone) { GoodData::Project.create(title: 'import clone test', client: @client, auth_token: ConnectionHelper::SECRETS[:gd_project_token]) }
```
Comment (yenkins) [inline spec/integration/models/project_spec.rb]: warning: Similar code found in project_spec.rb:164, project_spec.rb:208 (mass = 32) (flay)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/538/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1375/ : FAILURE in 2m 20s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1645/ : SUCCESS in 2m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/539/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1376/ : SUCCESS in 1m 06s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1646/ : SUCCESS in 1m 05s

**Comment by panjan** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/540/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1501/ : FAILURE in 13m 22s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/72/ : SUCCESS in 24m 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/541/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1377/ : SUCCESS in 1m 36s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1647/ : SUCCESS in 1m 38s

**Comment by kubamahnert** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/126/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1378/ : SUCCESS in 3m 09s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1648/ : SUCCESS in 1m 15s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1503/ : SUCCESS in 13m 18s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/74/ : SUCCESS in 29m 40s

**Pull request merged**